app/controllers/application_controller.rb

```diff
@@ -1655,8 +1655,24 @@ def flash_error_and_goto_index(model, id)
     nil
   end
 
+  # Like find_or_goto_index, but allows redirect to a different index
+  def find_obj_or_goto_index(model:, obj_id:, index_path:)
+    model.safe_find(obj_id) ||
+      flash_obj_not_found_and_goto_index(
+        model: model, obj_id: obj_id, index_path: index_path
+      )
+  end
```

Comment:
I need something like this in order to be able to both:
- flash that an object of class A was not found, and
- redirect to the controller for class B.